src/Toolkit/src/Registry/RegistryFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Toolkit\Registry;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * @internal
+ *
+ * @author Jean-François Lépine
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final readonly class RegistryFactory
+{
+    public function __construct(
+        private ContainerInterface $registries,
```

Comment:
This is what I had in mind too except that these are tagged services that bundles add dynamically